tests/eibc_fulfillment_test.go

```diff
@@ -23,6 +24,381 @@ import (
+	require.NoError(t, err)
+	for i, eibcEvent := range eibcEvents {
+		fmt.Println(i, "EIBC Event:", eibcEvent)
+	}
+
+	var fulfill_demand_order = false
+	// fulfill demand orders from rollapp 1
+	for _, eibcEvent := range eibcEvents {
+		re := regexp.MustCompile(`^\d+`)
+		if re.ReplaceAllString(eibcEvent.Price, "") == rollappIBCDenom {
+			txhash, err := dymension.FullfillDemandOrder(ctx, eibcEvent.ID, marketMakerAddr)
+			require.NoError(t, err)
+			eibcEvent := getEibcEventFromTx(t, dymension, txhash)
+			if eibcEvent != nil {
+				fmt.Println("After order fulfillment:", eibcEvent)
+			}
+			fulfill_demand_order = true
+		}
+	}
+
+	if !fulfill_demand_order {
+		panic("No eibc demant order was fulfilled")
+	}
+
+	// wait a few blocks and verify sender received funds on the hub
+	err = testutil.WaitForBlocks(ctx, 5, dymension)
+	require.NoError(t, err)
+
+	// verify funds minus fee were added to receiver's address
+	balance, err = dymension.GetBalance(ctx, dymensionUserAddr, rollappIBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of dymensionUserAddr after fulfilling the order:", balance)
+	require.True(t, balance.Equal(transferAmountWithoutFee), fmt.Sprintf("Value mismatch. Expected %s, actual %s", transferAmountWithoutFee, balance))
+
+	// verify correct funds were deducted from market maker's wallet address
+	balance, err = dymension.GetBalance(ctx, marketMakerAddr, rollappIBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of marketMakerAddr after fulfilling the order:", balance)
+	expMmBalanceRollappDenom = expMmBalanceRollappDenom.Sub((transferAmountWithoutFee))
+	require.True(t, balance.Equal(expMmBalanceRollappDenom), fmt.Sprintf("Value mismatch. Expected %s, actual %s", expMmBalanceRollappDenom, balance))
+
+	// wait until packet finalization and verify funds + fee were added to market maker's wallet address
+	isFinalized, err = dymension.WaitUntilRollappHeightIsFinalized(ctx, rollapp1.GetChainID(), rollappHeight, 200)
+	require.NoError(t, err)
+	require.True(t, isFinalized)
+	balance, err = dymension.GetBalance(ctx, marketMakerAddr, rollappIBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of marketMakerAddr after packet finalization:", balance)
+	expMmBalanceRollappDenom = expMmBalanceRollappDenom.Add(transferDataRollapp1.Amount)
+	require.True(t, balance.Equal(expMmBalanceRollappDenom), fmt.Sprintf("Value mismatch. Expected %s, actual %s", expMmBalanceRollappDenom, balance))
+
+	// user should have received funds upon grace period of IBC packet from rollapp 2
+	err = testutil.WaitForBlocks(ctx, 30, rollapp2)
+	require.NoError(t, err)
+	balance, err = dymension.GetBalance(ctx, dymensionUserAddr, rollapp2IBCDenom)
+	require.NoError(t, err)
+	fmt.Println("Balance of dymensionUserAddr for rollapp 2 ibc denom after grace period:", balance)
+	require.True(t, balance.Equal(transferDataRollapp2.Amount), fmt.Sprintf("Value mismatch. Expected %s, actual %s", transferDataRollapp2.Amount, balance))
+
```

Comment:
We should add a check `no packet commitment left on the rollapp (i.e ack recieved). can be queries using the cli rollapp-evm q ibc channel packet-commitments transfer` here